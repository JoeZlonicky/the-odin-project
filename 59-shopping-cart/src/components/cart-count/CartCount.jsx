function CartCount({ cart }) {
  let count = 0;
  cart.forEach((value) => {
    count += value;
  });
  return <>{count}</>;
}

export default CartCount;
