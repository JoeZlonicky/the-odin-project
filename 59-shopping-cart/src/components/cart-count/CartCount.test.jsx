import { render } from '@testing-library/react';
import { expect, it } from 'vitest';
import CartCount from './CartCount';

const itemOne = {
  name: 'One',
  id: 1,
  img: '1.png',
  cost: 1,
};

const itemTwo = {
  name: 'Two',
  id: 2,
  img: '2.png',
  cost: 2,
};

const testCart = new Map([
  [itemOne, 3],
  [itemTwo, 5],
]);

it('Matches screenshot', () => {
  const { container } = render(<CartCount cart={testCart} />);
  expect(container).toMatchSnapshot();
});

it('Sums cart correctly', () => {
  const { container } = render(<CartCount cart={testCart} />);
  expect(container.textContent).toMatch(8);
});
