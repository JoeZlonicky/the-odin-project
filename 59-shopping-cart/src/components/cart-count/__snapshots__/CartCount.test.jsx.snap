// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Matches screenshot 1`] = `
<div>
  8
</div>
`;
